test/sequential/test-http2-ping-flood.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const http2 = require('http2');
+const net = require('net');
+const http2util = require('../common/http2');
+
+// Test that ping flooding causes the session to be torn down
+
+const kSettings = new http2util.SettingsFrame();
+const kPing = new http2util.PingFrame();
+
+const server = http2.createServer();
+
+server.on('stream', common.mustNotCall());
+server.on('session', common.mustCall((session) => {
+  session.on('error', common.expectsError({
+    code: 'ERR_HTTP2_ERROR',
+    type: Error,
+    message:
+      'Flooding was detected in this HTTP/2 session, and it must be closed'
+  }));
+  session.on('close', common.mustCall(() => {
+    server.close();
+  }));
+}));
+
+server.listen(0, common.mustCall(() => {
+  const client = net.connect(server.address().port);
+
+  // nghttp2 uses a limit of 10000 items in it's outbound queue.
+  // If this number is exceeded, a flooding error is raised. Set
+  // this lim higher to account for the ones that nghttp2 is
+  // successfully able to respond to.
+  client.on('connect', common.mustCall(() => {
+    client.write(http2util.kClientMagic, () => {
+      client.write(kSettings.data, () => {
+        for (let n = 0; n < 35000; n++)
+          client.write(kPing.data);
+      });
+    });
+  }));
+
+  // We don't care if an error is emitted here, but don't crash
+  client.on('error', () => {});
```

Comment:
can you make this a proper function?

Reply:
In which sense? `client.on('error', function() {})`  ? I can, but I'm curious about why it matters in this case.

Reply:
I mean, handling/checking the error, you answered on another file. Why don't you put a note about the fact that the error can or cannot happen depending on the OS?

Reply:
Already done :-)